### PR TITLE
Added support for defining provider parameters in rebar.config

### PR DESCRIPTION
**Pull request by fadushin**

### CODE REVIEW

File: examples/mylib/rebar.config [hunk 1]
```diff
@@ -3,3 +3,13 @@
 {plugins, [
     {atomvm_rebar3_plugin, {git, "https://github.com/fadushin/atomvm_rebar3_plugin.git", {branch, "master"}}}
 ]}.
```
Comment (fadushin) [inline examples/mylib/rebar.config:5]: Fix “fadushin” ☝🏽

File: src/legacy_stm32_flash_provider.erl [hunk 2]
```diff
@@ -24,7 +24,7 @@
 -export([init/1, do/1, format_error/1]).
 
 -define(PROVIDER, stm32_flash).
--define(DEPS, [packbeam]).
+-define(DEPS, [{default, app_disovery}]).
```
Comment (fadushin) [inline src/legacy_stm32_flash_provider.erl:27]: Bad rebase.  Fix.

**Review by UncleGrumpy — COMMENTED** [review]
A few small corrections, and suggestions.  I also threw in a few comments about future improvements, just so I wouldn't forget to mention them later.

File: README.md [hunk 3]
```diff
@@ -1,104 +1,125 @@
 # atomvm_rebar3_plugin
 
-A `rebar3` plugin for simplifying development of Erlang applications targeted for the <a href="http://github.com/bettio/AtomVM">AtomVM</a> Erlang abstract machine.
+A [`rebar3`](https://rebar3.org) plugin for simplifying development of Erlang applications targeted for the <a href="http://github.com/bettio/AtomVM">AtomVM</a> Erlang abstract machine.
```
[the remark this answers is not in the document]
Comment (fadushin) [reply README.md]: Fixed

File: README.md [hunk 4]
```diff
@@ -1,104 +1,125 @@
...
+    ===> Verifying dependencies...
+    ===> Compiling myapp
+    ===> AVM file written to .../myapp/_build/default/lib/myapp.avm
+
+An AtomVM AVM file named `myapp.avm` is created in the `_build/default/lib` directory:
+
+    shell$ ls -l _build/default/lib/myapp.avm
+    -rw-rw-r--  1 joe  wheel  328 Jan 1 1970 00:01 _build/default/lib/myapp.avm
+
+If you have it installed, you can use the [`packbeam`](https://github.com/atomvm/atomvm_packbeam) tool to list the contents of this generated file:
```
[the remark this answers is not in the document]
Comment (fadushin) [reply README.md:42]: Ask and ye shall receive

File: README.md [hunk 5]
```diff
@@ -296,28 +345,44 @@ By default, the `stm32_flash` target will assume the `st-flash` command is avail
...
+
+The `stm32_flash` task depends on the `packbeam` task, so the packbeam file will get automatically built if any changes have been made to its dependencies.
 
-## The `pico_flash` target
+### The `pico_flash` task
+
+#### Flashing an application to a pico (rp2040) device
 
-### Flashing an application to a pico (rp2040) device
 You may use the `pico_flash` target to copy the generated AtomVM packbeam application in uf2 format to the flash storage on an Pico device connected to usb. It is not necessary to push the `BOOTSEL` button while plugging in the Pico to usb, instead provide the path of the device to reset. On Linux this is typically `/dev/ttyACM0` (the same device used to monitor serial), on MacOS it is a cu.usbmodem device matching `/dev/cu.usbmodem14*` (not the /dev/tty.usbmodem14___ device used for serial monitoring).
```
[the remark this answers is not in the document]
Comment (fadushin) [reply README.md]: Fixed

File: README.md [hunk 6]
```diff
@@ -351,43 +416,83 @@ If your pico uses a different device path or mount directory supply the full pat
...
+
+Alternatively, the following environment variables may be used to control the above settings:
+
+* ATOMVM_REBAR3_PLUGIN_PICO_MOUNT_PATH
+* ATOMVM_REBAR3_PLUGIN_PICO_RESET_DEV
+
+Any setting specified on the command line take precedence over entries in `rebar.config`, which in turn take precedence over environment variable settings, which in turn take precedence over the default values specified above.
 
-The `uf2create` target is used to generated an uf2 binary suitable for running on a Pico (RP2040) device from an AtomVM packbeam (`.avm`) file.
+The `pico_flash` task depends on the `packbeam` task, so the packbeam file will get automatically built if any changes have been made to its dependencies.
```
[the remark this answers is not in the document]
Comment (fadushin) [reply README.md]: Fixed

File: README.md [hunk 7]
```diff
@@ -250,32 +293,38 @@ Alternatively, the following environment variables may be used to control the ab
...
+You may use the `stm32_flash` task to flash the generated AtomVM packbeam application to the flash storage on an STM32 device connected to an st-link.
+
+#### Preparing an application for flashing
+
+The stm32 builds of AtomVM do not include a library partition and the [AtomVM](https://github.com/atomvm/AtomVM) `atomvmlib.avm` core library is not flashed to the device. Instead the application should be compiled and packed along with `atomvmlib.avm` before flashing, for example:
 
     shell$ rebar3 atomvm packbeam -p -e /path/to/atomvmlib.avm
 
-Consult `rebar3 help packbeam` for other options.
+You can acquire the latest [AtomVM](https://github.com/atomvm/AtomVM) `atomvmlib.avm` core library from the [Release](https://github.com/atomvm/AtomVM/releases) section of the [AtomVM](https://github.com/atomvm/AtomVM) Github repository.
```
[the remark this answers is not in the document]
Comment (fadushin) [reply README.md:321]: Why is that?  We are shipping `atomvmlib.avm`, right?  What binaries are we talking about here -- binaries for STM?

File: priv/rebar.config [hunk 8]
```diff
@@ -1,5 +1,25 @@
+%%
+%% Copyright (c) {{copyright_year}} <{{author_email}}>
```
[the remark this answers is not in the document]
Comment (fadushin) [reply priv/rebar.config:2]: I think this pattern is fine -- it's what we use everywhere else.

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (UncleGrumpy) [reply README.md:321]: Yes, exactly. The stm32 AtomVM binary.

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (UncleGrumpy) [reply README.md:42]: Sweet!

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (fadushin) [reply README.md]: Added a note to the README about matching.  I think we should work on a release for STM32, maybe just the Nucleo board.

**Review by UncleGrumpy — APPROVED** [review]

**Pull request merged**